src/main/integrations/browser-window-session.ts

```diff
@@ -0,0 +1,97 @@
+import { Integration } from '@sentry/types';
+import { app, BrowserWindow } from 'electron';
+
+import { ELECTRON_MAJOR_VERSION } from '../electron-normalize';
+import { endSession, endSessionOnExit, startSession } from '../sessions';
+
+interface Options {
+  /**
+   * Number of seconds to wait before ending a session after the app loses focus.
+   *
+   * Default: 10 seconds
+   */
+  backgroundTimeoutSeconds?: number;
+}
+
+// The state can be, active, inactive, or waiting for a timeout
+type SessionState = 'active' | 'inactive' | { timer: NodeJS.Timeout };
```

Comment:
I prefer if this was a discriminated union so we don't have to have to typeof checks.

```ts
type SessionState =
  | {
      type: 'active';
    }
  | {
      type: 'inactive';
    }
  | {
      type: 'waiting';
      timer: NodeJS.Timeout;
    };
```

Reply:
Yes good idea